File: rock_paper_scissors.py
import random

def runGame():
    player = True
    computerChoices = ["Rock", "Paper", "Scissors"]
    computerStr = "Computer says: "
    computerAnswer = random.choice(computerChoices)
    playerAnswer = input("Rock, Paper or Scissors?: ").lower()
    print("\n")
    
    while player == (True):
        if playerAnswer == "rock" and computerAnswer == ("Rock"):
            print(computerStr + computerAnswer)
            print("It's a tie!")
        elif playerAnswer == "paper" and computerAnswer == ("Paper"):
            print(computerStr + computerAnswer)
            print("It's a tie!")
        elif playerAnswer == "scissors" and computerAnswer == ("Scissors"):
            print(computerStr + computerAnswer)
            print("It's a tie!")
        elif playerAnswer == "rock" and computerAnswer == ("Scissors"):
            print(computerStr + computerAnswer)
            print("You win!")
        elif playerAnswer == "paper" and computerAnswer == ("Rock"):
            print(computerStr + computerAnswer)
            print("You win!")
        elif playerAnswer == "scissors" and computerAnswer == ("Paper"):
            print(computerStr + computerAnswer)
            print("You win!")
        else:
            print(computerStr + computerAnswer)
            print("You Lose.")
        player = False

def restart():
    retry = input("Again?(Y/N): ").lower()
    if retry == "y":
        runGame()
    else:
        exit()

runGame()
while True:
    restart()

File: rock_paper_scissors.rb
def runGame()
    player = true
    computerChoices = ["Rock", "Paper", "Scissors"]
    computerStr = "Computer says: "
    computerAnswer = computerChoices.sample
    playerStr = "Rock, Paper or Scissors?: "
    puts playerStr
    playerAnswer = gets.downcase().strip()
    puts "\n"

    while player == true
        if (playerAnswer == "rock" && computerAnswer == "Rock")
            puts computerStr + computerAnswer
            puts "\n"
            puts "It's a tie!"
        elsif (playerAnswer == "paper" && computerAnswer == "Paper")
            puts computerStr + computerAnswer
            puts "\n"
            puts "It's a tie!"
        elsif (playerAnswer == "scissors" && computerAnswer == "Scissors")
            puts computerStr + computerAnswer
            puts "\n"
            puts "It's a tie!"
        elsif (playerAnswer == "rock" && computerAnswer == "Scissors")
            puts computerStr + computerAnswer
            puts "\n"
            puts "You win!"
        elsif (playerAnswer == "paper" && computerAnswer == "Rock")
            puts computerStr + computerAnswer
            puts "\n"
            puts "You win!"
        elsif (playerAnswer == "scissors" && computerAnswer == "Paper")
            puts computerStr + computerAnswer
            puts "\n"
            puts "You win!"
        else
            puts computerStr + computerAnswer
            puts "\n"
            puts "You Lose."
        end
        player = false
    end
end

def restart()
puts "Again?(Y/N): "
    start_over = gets.downcase().strip()
    if (start_over == "y")
        runGame()
    else
        exit()
    end
end
runGame()
while true
    restart()
end
